internal/cli/atlas/backup/compliancepolicy/setup.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compliancepolicy
+
+import (
+	"context"
+	"errors"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/file"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
+)
+
+type SetupOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	policy  *atlasv2.DataProtectionSettings
+	store   store.CompliancePolicy
+	fs      afero.Fs
+	path    string
+	confirm bool
+}
+
+var setupTemplate = `
+Your backup compliance policy has been set up with the following configuration:
+
+Project:	{{.ProjectId}}
+Authorized e-mail:	{{.AuthorizedEmail}}
+Copy protection enabled:	{{.CopyProtectionEnabled}}
+Encryption at rest enabled:	{{.EncryptionAtRestEnabled}}
+Point-in-Time restores enabled:	{{.PitEnabled}}
+Restore window days:	{{.RestoreWindowDays}}
+
+POLICIES
+ID	FREQUENCY INTERVAL	FREQUENCY TYPE	RETENTION
+{{- range .ScheduledPolicyItems}}
+{{.Id}}	{{if eq .FrequencyType "hourly"}}{{.FrequencyInterval}}{{else}}-{{end}}	{{.FrequencyType}}	{{.RetentionValue}} {{.RetentionUnit}}
+{{- end}}
+{{if .OnDemandPolicyItem}}{{.OnDemandPolicyItem.Id}}	-	{{.OnDemandPolicyItem.FrequencyType}}	{{.OnDemandPolicyItem.RetentionValue}} {{.OnDemandPolicyItem.RetentionUnit}}{{end}}
+`
+
+func (opts *SetupOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *SetupOpts) setupWatcher() (bool, error) {
+	res, err := opts.store.DescribeCompliancePolicy(opts.ConfigProjectID())
+	opts.policy = res
+	if err != nil {
+		return false, err
+	}
+	if res.GetState() == "" {
+		return false, errors.New("could not access State field")
+	}
+	return (res.GetState() == active), nil
+}
+
+func (opts *SetupOpts) Run() error {
+	opts.policy.SetProjectId(opts.ConfigProjectID())
```

Comment:
[q] is this necessary? it should be given by the path?
if yes I think youc an encapsulate this logic in the store

Reply:
I think there is a decision to be made. To we prioritise the projectId set in the json file, or the one in the config? My initial thought was the one in the config because then we are consistent with how the rest of the `compliancepolicy` commands use `projectId`.

Reply:
This goes beyond a decision we could make, for example what does the API do whit this field, is it possible that in reality this field is readonly?, what happens when you make a request and this field is not set, what happens if you set it to a different ID from the one in the path?

Reply:
>what happens when you make a request and this field is not set

We get this response:
```json
"detail": "Invalid Backup Compliance Policy settings: [projectId] cannot be null.",
    "error": 400,
    "errorCode": "BACKUP_COMPLIANCE_POLICY_SETTINGS_INVALID",
    "parameters": [
        "[projectId] cannot be null"
    ],
    "reason": "Bad Request"
```

>what happens if you set it to a different ID from the one in the path?

It seems to me that it's ignored (unless it's in the incorrect format (e.g. 3 characters long)). I guess this means that it is read-only? 

In the context of the CLI, we just pass a `projectId` and these json/path details are hidden from us. Which takes me back to my original question:
Let's say the user passes a `--projectId` _and_ there is a `projectId` field in the json that they provided. Which one takes preference? What if `--projectId` is not present?

Reply:
Let's start with the easy one
>What if --projectId is not present?

`opts.ConfigProjectID()` is layered, it looks at `--projectId`, the env `MONGODB_ATLAS_PROJECT_ID`, or the config file value for `project_id` of the selected profile

now a harder one
>Let's say the user passes a --projectId and there is a projectId field in the json that they provided.

the request is 
```sh
curl https://cloud.mongodb.com/api/atlas/v2/groups/<projectId>/backupCompliancePolicy --data "{ "projectId": <projectId>, ...(other fields omitted for simplicity) }
```

you have 2 project Ids one on the url and one on the body, this is my original question
>[q] is this necessary? it should be given by the path?
if yes I think you can encapsulate this logic in the store

What I would've liked to confirm is what happens if the ID on the path is different from the ID on the body, I did this work for you, it is validated and has to be the same 

Which yes, let's encapsulate the logic on the project ID to the store, the value on the body is not meaningful only the one ont he path matters specially as int he cli you can only provide the path id via
>`--projectId`, the env `MONGODB_ATLAS_PROJECT_ID`, or the config file value for `project_id`

Reply:
Fixed